src/civis/utils/_jobs.py

```diff
@@ -96,3 +104,149 @@ def run_template(id, arguments, JSONValue=False, client=None):
     else:
         file_ids = {o.name: o.object_id for o in outputs}
         return file_ids
+
+
+def _timestamp_from_iso_str(s):
+    """Return an integer POSIX timestamp for a given ISO date string.
+
+    Note: Until Python 3.11, datetime.fromisoformat doesn't work
+    with the format returned by Civis Platform.
+    """
+    try:
+        return datetime.fromisoformat(s).timestamp()
+    except ValueError:
+        try:
+            # This is the format that Civis Platform returns.
+            return datetime.strptime(s, "%Y-%m-%dT%H:%M:%S.%f%z").timestamp()
+        except ValueError:
+            # Another format, just in case.
+            return datetime.strptime(s, "%Y-%m-%dT%H:%M:%S%z").timestamp()
+
+
+def _compute_effective_max_log_id(logs):
```

Comment:
There's fairly similar logic in our internal react code for the log display component of the scripts interface.